FirebaseUI/Auth/AuthProviderUI/Google/Source/FIRGoogleAuthUI.m

```diff
@@ -60,17 +65,22 @@ @implementation FIRGoogleAuthUI {
   FIRAuthProviderSignInCompletionBlock _pendingSignInCallback;
 }
 
-- (nullable instancetype)init {
+- (instancetype)init {
   @throw [NSException exceptionWithName:@"Attempt to call unavailable initializer."
                                  reason:@"Please call the designated initializer."
                                userInfo:nil];
 }
 
-- (nullable instancetype)initWithClientID:(NSString *)clientID {
+- (instancetype)initWithClientID:(NSString *)clientID {
+  return [self initWithClientID:clientID
+                         scopes:@[kGoogleUserInfoEmailScope, kGoogleUserInfoProfileScope]];
+}
+
+- (instancetype)initWithClientID:(NSString *)clientID scopes:(NSArray *)scopes {
   self = [super init];
   if (self) {
     _clientID = [clientID copy];
-    _scopes = @[ @"email", @"profile" ];
+    _scopes = scopes;
```

Comment:
This is a bit of an example of the concern from before - you're assigning to scopes, and skipping the "copy" property attribute. I think the right thing to do is to make the public property nonatomic, readonly, and in the anonymous category in the .m file override it to nonatomic, copy, readwrite.